cvat/apps/engine/views.py

```diff
@@ -475,6 +477,23 @@ def data(self, request, pk):
 
                 elif data_type == 'preview':
                     return sendfile(request, frame_provider.get_preview())
+
+                elif data_type == 'context_image':
+                    if db_task.dimension == DimensionType.DIM_3D:
+                        data_id = int(data_id)
+                        image = Image.objects.get(data_id=db_task.data_id, frame=data_id)
+                        for i in image.related_files.all():
+                            path = os.path.realpath(str(i.path))
+                            image = PIL.Image.open(path)
+                            buf = io.BytesIO()
+                            image.save(buf, format='JPEG', optimize=True)
```

Comment:
Consider using opencv here, because it's significantly faster.

Reply:
Made changes accordingly.